### PR TITLE
Bump version to 1.2.0-rc1

### PR DESCRIPTION
**Pull request by dongluochen**
Signed-off-by: Dong Chen <dongluo.chen@docker.com>

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,43 @@
 # Changelog
 
+## 1.2.0-rc1 (2016-03-22)
+
+#### Scheduler
+
+- Move rescheduling out of exprimental
```
Comment (nishanttotla) [inline CHANGELOG.md]: Typo: `exprimental` -> `experimental`

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,43 @@
...
+- Differentiate constraint errors from affinity errors
+- Printing unsatisfiable constraints for container scheduling failure
+
+#### API
+
+- Introduce engine-api client to Swarm
+- Implement 'info' and 'version' with engine-api
+- Plumbing context through all calls to engine-api client
```
Comment (nishanttotla) [inline CHANGELOG.md]: We don't really use this, so should we mention it in the changelog?

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,43 @@
...
+
+- Support rescind offer in swarm
+- Update mesos tests
+
+#### Misc
+
+- Fix HostConfig for /start endpoint
+- Skip redundant endpoints in "network inspect"
+- Remove parameter which is not used in createDiscovery
+- Validate duration flags:--delay, --timeout,--replication-ttl
```
Comment (nishanttotla) [inline CHANGELOG.md]: nit: space after comma

File: CHANGELOG.md (hunk 2 above)
Comment (dongluochen) [inline CHANGELOG.md]: I'll remove it. I think it's also included in "Introduce engine-api client to Swarm".

**Comment by nishanttotla** [comment]
LGTM

**Comment by vieux** [comment]
LGTM

**Pull request merged**